File: sast-python3/case/completeness/single_app_tracing/control_flow/conditional_stmt/conditional_match_001_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->流程控制语句->条件语句
# scene introduction = match
# level = 2
# bind_url = completeness/single_app_tracing/control_flow/conditional_stmt/conditional_match_001_T
# evaluation information end


import os

def conditional_match_001_T(taint_src):
    value = taint_src  
    match 2:
        case 2:  
            taint_sink(value)

def taint_sink(o):
    os.system(o)  

if __name__ == "__main__":
    taint_src = "taint_src_value"
    conditional_match_001_T(taint_src)

File: sast-php/case/completeness/single_app_tracing/control_flow/conditional_stmt/conditional_match_001_T.php
<?php
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用追踪->控制流
// scene introduction = PHP 8 match 表达式传递污点
// level = 2
// bind_url = completeness/single_app_tracing/control_flow/conditional_stmt/conditional_match_001_T
// evaluation information end

function conditional_match_001_T($__taint_src) {
    $type = 1;
    $result = match($type) {
        1 => $__taint_src,
        2 => "safe_value",
        default => "default_safe",
    };
    __taint_sink($result);
}

function __taint_sink($o) {
    shell_exec($o);
}

$taint_src = "taint_src_value";
conditional_match_001_T($taint_src);
